cpp/src/parquet/arrow/reader.cc

```diff
@@ -216,1462 +184,488 @@ class RowGroupRecordBatchReader : public ::arrow::RecordBatchReader {
 };
 
 // ----------------------------------------------------------------------
-// File reader implementation
+// FileReaderImpl forward declaration
```

Comment:
It's not really a "forward declaration", right? Seems like an implementation to me :-)